Reverse order of Transmission Risk Levels used for Diagnosis Keys Upload (closes #678)

mh- opened the pull request:
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [x] Make sure that your PR does not contain changes in text files, therefore the PR must not contain changes in values/strings/* and / or assets/* (see issue #332 for further information)
* [x] Make sure that your PR does not contain compiled sources (already set by the default .gitignore) and / or binary files

## Description

Fixes https://github.com/corona-warn-app/cwa-app-android/issues/678

htooisap committed:
Merge pull request #347 from corona-warn-app/dev

0.8.14

jakobmoellerdev committed:
Merge pull request #354 from corona-warn-app/dev

0.8.15

jakobmoellerdev committed:
Merge pull request #362 from corona-warn-app/dev

0.8.15 Hotfix

jakobmoellerdev committed:
Merge pull request #367 from corona-warn-app/dev

0.8.16

jakobmoellerdev committed:
Merge pull request #385 from corona-warn-app/dev

0.8.17

jakobmoellerdev committed:
Merge pull request #390 from corona-warn-app/dev

0.8.17

jakobmoellerdev committed:
Merge pull request #400 from corona-warn-app/dev

0.8.18

jakobmoellerdev committed:
Merge pull request #437 from corona-warn-app/dev

1.0.0

jakobmoellerdev committed:
Merge pull request #439 from corona-warn-app/dev

1.0.1

harambasicluka committed:
Merge pull request #445 from corona-warn-app/dev

1.0.1

jakobmoellerdev committed:
Merge pull request #447 from corona-warn-app/dev

1.0.1

jakobmoellerdev committed:
Merge pull request #470 from corona-warn-app/dev

1.0.2

jakobmoellerdev committed:
Merge pull request #663 from corona-warn-app/dev

1.0.3

mh- committed:
Reverse order of the "Transmission Risk Levels" profile for Diagnosis Keys upload
Also added test for ProtoFormatConverterExtensions.kt

mh- committed:
Merge branch 'master' of https://github.com/mh-/cwa-app-android

ChristopherSchmitz commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ProtoFormatConverterExtensions.kt:
```diff
@@ -33,9 +33,9 @@ object ProtoFormatConverterExtensions {
 
     fun List<TemporaryExposureKey>.transformKeyHistoryToExternalFormat() =
-        this.sortedWith(compareBy { it.rollingStartIntervalNumber })
+        this.sortedWith(compareBy { it.rollingStartIntervalNumber }).asReversed()
```
@mh- next to the asReversed option I see the other option to use "compareByDescending" instead of "compareBy". This option would result into better readability of code.

Do you see anything speaking against the usage of "compareByDescending"? (for sure we remove as Reversed in all the places)

mh- replied:
@ChristopherSchmitz `compareByDescending` is probably better, if it makes the code more readable. I don't read Kotlin a lot and indeed had a hard time understanding the code. So I just copied the pattern from [.limitKeyCount()](https://github.com/corona-warn-app/cwa-app-android/blob/c2e5bd49093193060dded4d1545f783a498a96da/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ProtoFormatConverterExtensions.kt#L33) and wrote a test case to see what it's actually doing. 
The test case assumes that the keys are ordered chronologically, but if that's not required for upload to the server, we could also move the 2nd sorting to the test.
Not sure how you prefer to proceed? Do you want to edit the PR, or should I do it?

ChristopherSchmitz replied:
@mh since we use on iOS a desc order when submitting to the server I would keep the behaviour consistent.
If you like to adjust your PR this would be wonderful. We plan to close and merge this in the next 2 hours :-)

mh- replied:
@ChristopherSchmitz done.

mh- committed:
Reverse order of the "Transmission Risk Levels" profile for Diagnosis Keys upload
Small adjustments: 1) Use compareByDescending instead of compareBy and asReversed();
2) leave the list for upload in a reversed chronological order, just like in the iOS app;
3) add sorting to the test to keep the desired values in the test readable in chronological order

ChristopherSchmitz reviewed (approved):
@mh- many thanks for your support! This looks really good.
